pkg/storage/utils/decomposedfs/upload.go

```diff
@@ -383,12 +397,58 @@ func (fs *Decomposedfs) GetUpload(ctx context.Context, id string) (tusd.Upload,
 	return &fileUpload{
 		info:     info,
 		binPath:  info.Storage["BinPath"],
-		infoPath: infoPath,
+		infoPath: filepath.Join(fs.o.Root, "uploads", id+".info"),
 		fs:       fs,
 		ctx:      ctx,
 	}, nil
 }
 
+// ListUploads returns a list of all incomplete uploads
+func (fs *Decomposedfs) ListUploads() ([]tusd.FileInfo, error) {
+	infos, err := filepath.Glob(filepath.Join(fs.o.Root, "uploads", "*.info"))
+	if err != nil {
+		return nil, err
+	}
+	list := make([]tusd.FileInfo, len(infos))
+	idRegexp := regexp.MustCompile(".*/([^/]+).info")
+	for i, info := range infos {
+		match := idRegexp.FindStringSubmatch(info)
+		upload, err := fs.readInfo(match[1])
```

Comment:
Could we run into a panic here? `index out of range`?

Reply:
I can't see how since the list is a glob of `uploads/*.infos` and the regexp always matches those results. I can add a safety check though if you prefer, it shouldn't hurt.

Reply:
No you are probably right. The only thing I could imagine was a faulty upload with empty ID leading to `uploads/.info` which wouldn't be found by the regexp. But I'll leave assessing ciritically to you, I'm fine with leaving as is.